Deliverable 1/Subdivisions/requirements_review.tex

```diff
@@ -1 +1,196 @@
+
+\medskip
+
+Possible Alternatives:
+\begin{itemize}
+  \item[] “Support at least 500 concurrent users sending queries within a 10 second window.”
+\end{itemize}
+
+\subsubsection*{Requirement 63}
+\begin{itemize}
+  \item[(63)] "Search results must never be empty, i.e., not possible for 'no results found'”
+\end{itemize}
+
+Depending on how this requirement is read it can be interpreted multiple ways. Does this mean to say when no results are found there will be a specific “no results found page”?
+Or does it mean to say that there must be results pulled no matter what the query is?
+
+\medskip
+
+Possible Alternatives:
+\begin{itemize}
+  \item[] “A query must result either return a “no results found page” or a page with the corresponding search results.”
+  \item[] “A query must always return search results. If there are no immediately clear search results, the search parameters must be broadened until results are found.”
+\end{itemize}
+
+
+\subsubsection*{Requirement 66}
+\begin{itemize}
+  \item[(66)] "Ranking should consider search history”
+\end{itemize}
+
+Search history can be individual or global. Which does this requirement refer to? If we follow (59), its global, but if we follow (16), its individual.
+
+\medskip
+
+Possible Alternatives:
+\begin{itemize}
+  \item[] “Ranking will consider a user's search history”
+  \item[] “Ranking will consider the full Search Engine history”
+\end{itemize}
+
+\subsection*{Missing Requirements}
+\subsubsection{Implicit Requirements}
+\begin{itemize}
+  \item Create program logs from all saveable data produced from the program
+  \item Log when URL web pages are updated
+  \item Administration can access all logs through command lines to ensure maintainability
+  \item When a user clicks on a search result, open the webpage in another tab or window
+  \item There is a search bar where the user can type their query
+  \item Typing in the search bar should have the current query show in the search bar
+  \item After a query is entered, show the first 10 results on the first page
+  \item If there are more than 10 results, allow users to look through all the results by clicking a button to show more results
+  \item Crawling should stop when data storage is full
+\end{itemize}
+\subsubsection{Other Requirements}
+\begin{itemize}
+  \item Store a list of blacklisted websites to exclude from crawling
+  \item Allow for the creation of anonymous profiles for each user to support ranking based on personal data
+\end{itemize}
+
+\subsection*{Avoiding Marginalization}
+\begin{itemize}
+  \item Ensure the results that show up first are credible sources
+  \item Add a disclaimer regarding the uncertainty of untrustworthy sites
+  \item Queries with typos or slang should have similar results to similar "well formed" queried
+  \item Queries in different languages should have similar results to similar english queries
```

Comment:
Queries in other supported languages should have similar results to their equivalent english queries

Reply:
don't want to have ambiguous language, especially since one of the requirements that we flagged said "other languages"